src/WebJobs.Extensions.DurableTask/DurableTaskJobHostConfigurationExtensions.cs

```diff
@@ -107,6 +109,33 @@ public static IWebJobsBuilder AddDurableTask(this IWebJobsBuilder builder)
             return builder;
         }
 
+#if FUNCTIONS_V3_OR_GREATER
+        /// <summary>
+        /// Adds the IScaleMonitor and ITargetScaler providers for the Durable Triggers.<see cref="IWebJobsBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IWebJobsBuilder"/> to configure.</param>
+        /// <returns>Returns the provided <see cref="IWebJobsBuilder"/>.</returns>
+        public static IWebJobsBuilder AddDurableScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
+        {
+            IServiceProvider serviceProvider = null;
+            var scalerProvider = new Lazy<DurableTaskTriggersScaleProvider>(() => new DurableTaskTriggersScaleProvider(serviceProvider, triggerMetadata));
+
+            builder.Services.AddSingleton<IScaleMonitorProvider>(resolvedServiceProvider =>
+            {
+                serviceProvider = serviceProvider ?? resolvedServiceProvider;
+                return scalerProvider.Value;
+            });
+
+            builder.Services.AddSingleton<ITargetScalerProvider>(resolvedServiceProvider =>
+            {
+                serviceProvider = serviceProvider ?? resolvedServiceProvider;
+                return scalerProvider.Value;
+            });
```

Comment:
The explanation for this lazy initialization pattern is here: https://github.com/Azure/azure-sdk-for-net/pull/34557/files#r1170772719

Reply:
Please change the pater to:
https://[github.com/Azure/azure-sdk-for-net/pull/38673](https://github.com/Azure/azure-sdk-for-net/pull/38673)

Reply:
Thanks, it's been addressed here: https://github.com/Azure/azure-functions-durable-extension/pull/2462/commits/3800635bc41957c9ec60880755dc5b4914d90868